nixos/tests/agda.nix

```diff
@@ -36,6 +36,8 @@ in
         "cp ${hello-world} HelloWorld.agda"
     )
     machine.succeed("agda -l standard-library -i . -c HelloWorld.agda")
+    # Check execution
+    machine.succeed('test "$(./HelloWorld)" = "Hello World!"')
```

Comment:
Maybe we should write the test like

```
assert "Hello World!" in machine.succeed(
  "./HelloWorld"
), "HelloWorld does not run properly"
```

which would be consistent with other test files.

Reply:
Perhaps, I am not an expert on testing, but if it works and gives a nice error message then that is surely an improvement

Reply:
It doesn't make much of a difference. Using `assert` we also check that `./HelloWorld` returns 0. `test` only validates the output of `./HelloWorld`.

test:

```
machine: must succeed: test "$(./HelloWorld)" = "Hello World!"
machine: output:
error:
Traceback (most recent call last):
  File "/nix/store/0fhfz90b5vc4k4gsk0gd6rj88m7gllz5-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 897, in run_tests
    exec(tests, globals())
  File "<string>", line 1, in <module>
  File "<string>", line 15, in <module>
  File "/nix/store/0fhfz90b5vc4k4gsk0gd6rj88m7gllz5-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 422, in succeed
    raise Exception(
Exception: command `test "$(./HelloWorld)" = "Hello World!"` failed (exit code 1)
```

assert:

```
machine: must succeed: ./HelloWorld
(0.02 seconds)
error:
Traceback (most recent call last):
  File "/nix/store/0fhfz90b5vc4k4gsk0gd6rj88m7gllz5-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 897, in run_tests
    exec(tests, globals())
  File "<string>", line 1, in <module>
  File "<string>", line 15, in <module>
AssertionError: HelloWorld does not run properly
```